packages/flutter_driver/lib/src/driver.dart

```diff
@@ -371,11 +371,8 @@ Future<VMServiceClientConnection> _waitAndConnect(String url) async {
         new rpc.Peer(new IOWebSocketChannel(ws2).cast())..listen()
       );
     } catch(e) {
-      if (ws1 != null)
-        ws1.close();
-
-      if (ws2 != null)
-        ws2.close();
+      await ws1?.close();
+      await ws2?.close();
```

Comment:
Not critical in this code. This is the case when connection failed and will be retried. Chances are calling close is a noop.